src/Runner.Common/JobServer.cs

```diff
@@ -45,8 +48,17 @@ public async Task ConnectAsync(VssConnection jobConnection)
                 }
                 catch (Exception ex) when (attemptCount > 0)
                 {
-                    Trace.Info($"Catch exception during connect. {attemptCount} attemp left.");
+                    Trace.Info($"Catch exception during connect. {attemptCount} attempts left.");
                     Trace.Error(ex);
+
+                    var configurationStore = HostContext.GetService<IConfigurationStore>();
```

Comment:
you might want to cache the config in the outer loop, so we don't need to read the file on every retry.